prov/gni/test/auth_key.c

```diff
@@ -76,3 +76,42 @@ Test(auth_key, failed_insert)
 	ret = _gnix_auth_key_create(NULL, 0);
 	cr_assert(ret == NULL, "unexpectedly created auth_key");
 }
+
+void *race_create_func(void *context)
+{
+	pthread_barrier_t *barrier = (pthread_barrier_t *) context;
+	struct gnix_auth_key *auth_key;
+
+	pthread_barrier_wait(barrier);
```

Comment:
Check return code.  I know we're not testing pthreads, but if something goes wrong, more info is better for diagnosis.